core/src/main/scala/kafka/admin/PreferredReplicaLeaderElectionCommand.scala

```diff
@@ -119,32 +118,21 @@ object PreferredReplicaLeaderElectionCommand extends Logging {
   }
 }
 
-class PreferredReplicaLeaderElectionCommand(zkUtils: ZkUtils, partitions: scala.collection.Set[TopicAndPartition])
+class PreferredReplicaLeaderElectionCommand(zkUtils: ZkUtils, partitionsFromUser: scala.collection.Set[TopicAndPartition])
   extends Logging {
   def moveLeaderToPreferredReplica() = {
     try {
-      val validPartitions = partitions.filter(p => validatePartition(zkUtils, p.topic, p.partition))
+      val topics = partitionsFromUser.map(_.topic).toSet
+      val partitionsFromZk = zkUtils.getPartitionsForTopics(topics.toSeq).flatMap{case (topic, partitions) =>
+        partitions.map(TopicAndPartition(topic, _))}.toSet
+
+      val (validPartitions, invalidPartitions) = partitionsFromUser.partition(p => partitionsFromZk.contains(p))
       PreferredReplicaLeaderElectionCommand.writePreferredReplicaElectionData(zkUtils, validPartitions)
+
+      info("Successfully started preferred replica election for partitions %s".format(validPartitions))
+      invalidPartitions.foreach(p => error("Skipping preferred replica leader election for partition %s since it doesn't exist.".format(p)))
```

Comment:
No, I don't understand the current script ignores missing partitions. I would prefer the script to just fail in that case.